framework/core/js/src/forum/components/AccessTokensList.tsx

```diff
@@ -0,0 +1,182 @@
+        </div>
+      );
+    }
+
+    items.add(
+      'createdAt',
+      <div className="AccessTokensList-item-createdAt">
+        <DataSegment label={app.translator.trans('core.forum.security.created')} value={humanTime(token.createdAt())} />
+      </div>
+    );
+
+    items.add(
+      'lastActivityAt',
+      <div className="AccessTokensList-item-lastActivityAt">
+        <DataSegment
+          label={app.translator.trans('core.forum.security.last_activity')}
+          value={
+            token.lastActivityAt()
+              ? [
+                  humanTime(token.lastActivityAt()),
+                  token.lastIpAddress() && [' — ', token.lastIpAddress()],
+                  this.attrs.type === 'token' && token.lastUserAgent() && [' — ', <span className="AccessTokensList-item-title-sub">{device}</span>],
+                ]
+              : app.translator.trans('core.forum.security.never')
+          }
+        />
+      </div>
+    );
+
+    return items;
+  }
+
+  tokenActionItems(token: AccessToken) {
+    const items = new ItemList<Mithril.Children>();
+
+    const deleteKey = {
+      session: 'terminate_session',
+      token: 'revoke_access_token',
+    }[this.attrs.type];
+
+    if (this.attrs.type === 'token') {
+      const isHidden = !this.showingTokens[token.id()!];
+      const displayKey = isHidden ? 'show_access_token' : 'hide_access_token';
+
+      items.add(
+        'toggleDisplay',
+        <Button
+          className="Button Button--inverted"
+          icon={isHidden ? 'fas fa-eye' : 'fas fa-eye-slash'}
+          onclick={() => {
+            this.showingTokens[token.id()!] = isHidden;
+            m.redraw();
+          }}
+        >
+          {app.translator.trans(`core.forum.security.${displayKey}`)}
+        </Button>
+      );
+    }
+
+    let revokeButton = (
+      <Button className="Button Button--danger" disabled={token.isCurrent()} loading={!!this.loading[token.id()!]} onclick={() => this.revoke(token)}>
+        {app.translator.trans(`core.forum.security.${deleteKey}`)}
+      </Button>
+    );
+
+    if (token.isCurrent()) {
+      revokeButton = (
+        <Tooltip text={app.translator.trans('core.forum.security.cannot_terminate_current_session')}>
+          <div tabindex="0">{revokeButton}</div>
+        </Tooltip>
+      );
+    }
+
+    items.add('revoke', revokeButton);
+
+    return items;
+  }
+
+  revoke(token: AccessToken) {
+    if (!confirm(extractText(app.translator.trans('core.forum.security.revoke_access_token_confirmation')))) return;
+
+    this.loading[token.id()!] = true;
+
+    return token.delete().then(() => {
+      this.loading[token.id()!] = false;
+      this.attrs.ondelete && this.attrs.ondelete(token);
+      const key = this.attrs.type === 'session' ? 'session_terminated' : 'token_revoked';
+      app.alerts.show({ type: 'success' }, app.translator.trans(`core.forum.security.${key}`, { count: 1 }));
+      m.redraw();
+    });
+  }
```

Comment:
I think an async function here is cleaner.

```suggestion
  async revoke(token: AccessToken) {
    if (!confirm(extractText(app.translator.trans('core.forum.security.revoke_access_token_confirmation')))) return;

    this.loading[token.id()!] = true;

    await token.delete()

    this.loading[token.id()!] = false;
    this.attrs.ondelete?.(token);
    const key = this.attrs.type === 'session' ? 'session_terminated' : 'token_revoked';
    app.alerts.show({ type: 'success' }, app.translator.trans(`core.forum.security.${key}`, { count: 1 }));

    m.redraw();
  }
```